### ISSUE TITLE
!!!SPOILER!!! Flag 3 Bug?

### ISSUE THREAD

**Issue by decidedlygray** [hackgnar/ble_ctf#4]
The hint for flag 3 was:

```
root@kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-read -a 0x0030
Characteristic value/descriptor: 4d 44 35 20 6f 66 20 44 65 76 69 63 65 20 4e 61 6d 6
root@kali-vm:~# echo -n "4d 44 35 20 6f 66 20 44 65 76 69 63 65 20 4e 61 6d 65" | tr -d ' ' | xxd -r -p
MD5 of Device Name
```

So retrieving the device name and md5sum'ing it, and submitting the first 20 characters doesn't get the flag:

```
root@kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-read --handle=0x0016
Characteristic value/descriptor: 32 62 30 30 30 34 32 66 37 34 38 31 63 37 62 30 35 36 63 34 62 34 31 30 64 32 38 66 33 33 63 66
root@kali-vm:~# echo -n "32 62 30 30 30 34 32 66 37 34 38 31 63 37 62 30 35 36 63 34 62 34 31 30 64 32 38 66 33 33 63 66" | tr -d " " | xxd -r -p
2b00042f7481c7b056c4b410d28f33cf
root@kali-vm:~# echo -n 2b00042f7481c7b056c4b410d28f33cf | md5sum
8489c638085eb7b7416e682af1dd5474  -
root@kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-write-req -a 0x002c -n $(echo -n "8489c638085eb7b7416e"|xxd -ps)
Characteristic value was written successfully
root@tuv-kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-read -a 0x002a|awk -F':' '{print $2}'|tr -d ' '|xxd -r -p;printf '\n'
Score:2 /20
```

However, ignoring the clue "MD5 of Device Name", and just returning exactly the hex version of the device name, not md5sum'ing anything, we get the flag:
```
root@kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-write-req -a 0x002c -n $(echo -n "2b00042f7481c7b056c4"|xxd -ps)
Characteristic value was written successfully
root@kali-vm:~# gatttool -b b4:E6:2d:96:14:7F --char-read -a 0x002a|awk -F':' '{print $2}'|tr -d ' '|xxd -r -p;printf '\n'
Score:3 /20
```

I think this is a bug? Or at the very least a misleading clue?

**Comment by vekkt0r** [comment]
I ran down this rabbit hole also, clue is ambiguous, but correct. Score is increasing because you are solving another flag, not flag 0x30.

**Comment by decidedlygray** [comment]
Thanks @vekkt0r! Your comment prompted me to review my other flags [and the code](https://github.com/hackgnar/ble_ctf/blob/9c7d52b4c6811f6fd2b94376e9051cb80d8e4757/main/gatts_table_creat_demo.c#L833). 

In case it helps anyone else, here is the problem I created for myself: when I first discovered the real answer for Flag 3, I did not limit what I sent to 20 characters (I sent the whole hash). I had actually solved Flag 3 with the answer for Flag 2. Then I continued on down a rabbit hole and ended up giving the answer for Flag 2  for Flag 3 (assuming the flags are in order in the code).
